pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -36,51 +37,56 @@ def __init__(
 
     def build_model(
         self,
-        data: pd.DataFrame,
-        adstock_max_lag: int = 4,
+        model_data: dict,
+        model_config: dict,
```

Comment:
Do we need to change the API?

Reply:
If we change that now we'll gain the flexibility to alter the initial params via the api using model_config), so far they were hardcoded (if we're sure that it will never need to be altered by users then indeed maybe it's not necessary)